projects/plugins/wpcomsh/frontend-notices/gifting-banner/gifting-banner.php

```diff
@@ -0,0 +1,268 @@
+	}
+
+	/**
+	 * Get how many days the banner should be dismissed based on plan type (30 days for monthly, 1 year for yearly)
+	 */
+	public function get_days_to_dismiss_banner() {
+		return false !== strpos( $this->current_plan->product_slug, 'monthly' ) ? 30 : 365;
+	}
+
+	/**
+	 * Get more info link.
+	 */
+	public function get_more_info_link() {
+		return 'https://wordpress.com/support/gift-a-wordpress-com-subscription/';
+	}
+
+	/**
+	 * Inject the gifting banner on WPCOM.
+	 */
+	public function inject_gifting_banner_wpcom() {
+		$days_to_expire             = ceil( ( strtotime( $this->current_plan->expiry_date ) - time() ) / DAY_IN_SECONDS );
+		$data                       = array();
+		$data['dismiss_days_count'] = $this->get_days_to_dismiss_banner();
+		$data['checkout_link']      = $this->get_checkout_link();
+		$data['more_info_link']     = localized_wpcom_url( $this->get_more_info_link() );
+		$data['i18n']               = array(
+			'title'       => $this->get_title_texts( $days_to_expire ),
+			'subtitle'    => $this->get_subtitle_texts( $this->current_plan, $days_to_expire ),
+			'button_text' => _x( 'Gift', 'verb', 'wpcomsh' ),
+		);
+		// Change the version if associated files are updated, current: 20230103.
+		wp_enqueue_style( 'gifting-banner', plugins_url( 'gifting-banner/css/gifting-banner.css', __FILE__ ), array(), '20230103' );
+		wp_enqueue_script( 'gifting-banner', plugins_url( 'gifting-banner/js/gifting-banner.js', __FILE__ ), array(), '20230103', true );
+		wp_localize_script( 'gifting-banner', 'gifting_banner', $data );
+		wp_set_script_translations( 'gifting-banner', 'wpcomsh' );
+	}
+
+	/**
+	 * Inject the gifting banner on WPCOMSH.
+	 */
+	public function inject_gifting_banner_wpcomsh() {
+		$days_to_expire             = ceil( ( strtotime( $this->current_plan->expiry_date ) - time() ) / DAY_IN_SECONDS );
+		$data                       = array();
+		$data['dismiss_days_count'] = $this->get_days_to_dismiss_banner();
+		$data['checkout_link']      = $this->get_checkout_link();
+		$data['more_info_link']     = $this->get_more_info_link();
+		$data['i18n']               = array(
+			'title'       => static::get_title_texts( $days_to_expire ),
+			'subtitle'    => static::get_subtitle_texts( $this->current_plan, $days_to_expire ),
+			'button_text' => _x(
+				'Gift',
+				'verb',
+				'wpcomsh'
+			),
+		);
+
+		wp_enqueue_style( 'gifting-banner', plugins_url( 'css/gifting-banner.css', __FILE__ ), array(), WPCOMSH_VERSION );
+		wp_enqueue_script( 'gifting-banner', plugins_url( 'js/gifting-banner.js', __FILE__ ), array(), WPCOMSH_VERSION, true );
+		wp_localize_script( 'gifting-banner', 'gifting_banner', $data );
+		wp_set_script_translations( 'gifting-banner', 'wpcomsh' );
+	}
+
+	/**
+	 * Get hosting plan purchase from the current site.
+	 * See: 1118-gh-Automattic/wpcomsh
```

Comment:
No need to shorthand this
```suggestion
	 * See: https://github.com/Automattic/wpcomsh/pull/1118
```